x/dsmr/node.go

```diff
@@ -200,28 +229,37 @@ func (n *Node[T]) BuildBlock(parent Block, timestamp int64) (Block, error) {
 
 	packer := wrappers.Packer{Bytes: make([]byte, 0, InitialChunkSize), MaxSize: consts.NetworkSizeLimit}
 	if err := codec.LinearCodec.MarshalInto(blk, &packer); err != nil {
-		return Block{}, err
+		return Block[T]{}, err
 	}
 
 	blk.blkBytes = packer.Bytes
 	blk.blkID = utils.ToID(blk.blkBytes)
 	return blk, nil
 }
 
-func (*Node[T]) Execute(ctx context.Context, _ Block, block Block) error {
+func (n *Node[T]) Execute(ctx context.Context, block Block[T]) error {
```

Comment:
I actually didn't understand why we had the parent parameter in the first place... I think if we wanted to keep this fn signature it should be a parent state instead of a block - or we accept the block to execute and assume that `Node` maintains parent state.